homeassistant/components/fyta/sensor.py

```diff
@@ -0,0 +1,171 @@
+        translation_key="fyta_salinity_status",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="temperature",
+        translation_key="fyta_temperature",
+        native_unit_of_measurement="°C",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:thermometer",
+    ),
+    FytaSensorEntityDescription(
+        key="light",
+        translation_key="fyta_light",
+        native_unit_of_measurement="mol/d",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:weather-sunny",
+    ),
+    FytaSensorEntityDescription(
+        key="moisture",
+        translation_key="fyta_moisture",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.MOISTURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:water-percent",
+    ),
+    FytaSensorEntityDescription(
+        key="salinity",
+        translation_key="fyta_salinity",
+        native_unit_of_measurement="mS/cm",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:sprout-outline",
+    ),
+    FytaSensorEntityDescription(
+        key="ph",
+        translation_key="fyta_PH",
+        device_class=SensorDeviceClass.PH,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:ph",
+    ),
+    FytaSensorEntityDescription(
+        key="battery_level",
+        translation_key="fyta_battery_level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:battery",
+    ),
+    FytaSensorEntityDescription(
+        key="last_updated",
+        translation_key="fyta_last_updated",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:update",
+    ),
+    FytaSensorEntityDescription(
+        key="plant_number",
+        translation_key="fyta_plant_number",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        icon="mdi:update",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the FYTA binary sensors."""
+    coordinator: FytaCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [
+            FytaSensor(coordinator, entry, sensor)
+            for sensor in SENSORS
+            if sensor.key in coordinator.data
+        ]
+    )
+
+    for plant_id in coordinator.plant_list:
+        async_add_entities(
+            [
+                FytaSensor(coordinator, entry, sensor, plant_id)
+                for sensor in SENSORS
+                if sensor.key in coordinator.data[plant_id]
+            ]
+        )
+
```

Comment:
Please only call async_add_entities once

Reply:
How could this best be achieved, as apparently the sensors for all plants are the same, so the plant_id has to be set in FytaSensor ?

Reply:
Just create a list and fill that with all entities and then call async_add_entities